### ISSUE TITLE
got a error result from two conflict regex

### ISSUE THREAD

**Issue by adiaixin** [cs-au-dk/dk.brics.automaton#9]
`String str = "num1=123&num2=456";
        String regex = "num1=.*&num2=.*";
        p = Pattern.compile(regex);
        System.out.println(p.matcher(str).matches());
        Automaton automaton = new RegExp(regex).toAutomaton();
        System.out.println(automaton.run(str));`
//result : true , false
if I add ad '\\' before '&' the result is "true, true"
is '&' a special  char in Automaton ? 
is it have other special char ?

**Comment by amoeller** [comment]
Yes, those characters have a special meaning. See the documentation: http://www.brics.dk/automaton/doc/index.html
in particular this constructor: http://www.brics.dk/automaton/doc/dk/brics/automaton/RegExp.html#RegExp-java.lang.String-int-
and the FAQ: http://www.brics.dk/automaton/faq.html
